src/init.c

```diff
@@ -750,6 +750,7 @@ static void init_global_mutexes(void) {
     JL_MUTEX_INIT(&global_roots_lock, "global_roots_lock");
     JL_MUTEX_INIT(&typecache_lock, "typecache_lock");
     JL_MUTEX_INIT(&profile_show_peek_cond_lock, "profile_show_peek_cond_lock");
+    jl_mutex_timing_lock_init();
```

Comment:
This seems to initialize this mutex strictly later than before (when it was in `jl_init_timing`)

Any reason why?

Reply:
I think it should be initialized with timing, not later - otherwise an arbitrary subset of timing won't be safe to use during the interim